response.go

```diff
@@ -1,31 +1,82 @@
 package epp
 
 import (
+	"bytes"
+	"encoding/xml"
 	"time"
 
-	ws "github.com/bombsimon/epp-server/types"
+	"github.com/bombsimon/epp-go/types"
 )
 
 // GreetResponse generates the greet command.
-func (s *Server) GreetResponse() ws.EppType {
-	return ws.EppType{
-		Greeting: &ws.GreetingType{
-			SvID: ws.SIDType{
-				Id: ws.ClIDType("asdb"),
+func (s *Server) GreetResponse() types.EPPGreeting {
+	return types.EPPGreeting{
+		Greeting: types.Greeting{
+			ServerID:   "default-server",
+			ServerDate: time.Now(),
+			ServiceMenu: types.ServiceMenu{
+				Version: []string{"1.0"},
 			},
-			SvDate: time.Now(),
 		},
 	}
 }
 
-// LoginResponse returns a successful login response.
-func (s *Server) LoginResponse() ws.EppType {
-	return ws.EppType{
-		Response: &ws.ResponseType{
-			TrID: ws.TrIDType{
-				ClTRID: ws.TrIDStringType("XXX-99999"),
-				SvTRID: ws.TrIDStringType("044"),
+// Encode will take a type that can be marshalled to XML, add the EPP staring
+// tag for all registered namespaces and return the XML as a byte slice.
+func (s *Server) Encode(data interface{}, ns map[string]string) ([]byte, error) {
```

Comment:
Could this be used by both the client and the service if it's not attached to the server?

Reply:
Yeah thought about that, that's a good idea! I think we should determine on a structure regarding names, multiple packages etc.